code/modules/surgery/organs/eyes.dm

```diff
@@ -251,15 +251,17 @@
 	if(!C || QDELETED(src) || QDELETED(user) || QDELETED(owner) || owner != user)
 		return
 	var/range = input(user, "Enter range (0 - [max_light_beam_distance])", "Range Select", 0) as null|num
-
-	set_distance(clamp(range, 0, max_light_beam_distance))
 	assume_rgb(C)
+	// Reactivate if eyes were already active for real time colour swapping!
+	if(active)
+		set_distance(clamp(range, 0, max_light_beam_distance))
+		activate(FALSE)
+	else
+		set_distance(clamp(range, 0, max_light_beam_distance))
```

Comment:
This is a duplicate line, move it out and then just do `if(active) activate(FALSE)`

Reply:
set_distance() would set active to false, but I can move the false assignment and it should work. Will do some testing before I nuke this dupe line.

Reply:
You can also just store the value beforehand, `var/old_active = active` for example.

Reply:
There, thanks for the advice!